pkg/kubectl/cmd/annotate.go

```diff
@@ -134,8 +132,8 @@ func NewCmdAnnotate(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().BoolVar(&options.all, "all", false, "select all resources in the namespace of the specified resource types")
 	cmd.Flags().StringVar(&options.resourceVersion, "resource-version", "", "If non-empty, the annotation update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
 	usage := "Filename, directory, or URL to a file identifying the resource to update the annotation"
-	kubectl.AddJsonFilenameFlag(cmd, &options.filenames, usage)
-	cmdutil.AddRecursiveFlag(cmd, &options.recursive)
+	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)
```

Comment:
pull these into a single function in cmdutil

Example: https://github.com/kubernetes/kubernetes/pull/25217/commits/a33c032f7d386df4dcfb006fa5b835a5fe543805